sympy/core/power.py

```diff
@@ -1736,6 +1736,11 @@ def mul(d1, d2):
         if not d.is_positive:
             g = g.simplify()
             _, d = g.leadterm(x)
+            if not d.is_positive:
+                if g.is_zero:
+                    return f**e
+                g = ((b - f)/f).expand()
+                _, d = g.leadterm(x)
             if not d.is_positive:
                 raise NotImplementedError()
```

Comment:
This block could be indented in the same way as lines 1742-3. It is not necessary to repeat the test `not d.is_positive` when d is positive after line 1738.